kernel/src/memop.rs

```diff
@@ -52,48 +52,48 @@ pub(crate) fn memop(process: &dyn Process, op_type: usize, r1: usize) -> Syscall
         // Op Type 1: SBRK
         1 => process
             .sbrk(r1 as isize)
-            .map(|addr| SyscallReturn::SuccessU32(addr as u32))
+            .map(|addr| SyscallReturn::SuccessPtr(addr))
             .unwrap_or(SyscallReturn::Failure(ErrorCode::NOMEM)),
 
         // Op Type 2: Process memory start
-        2 => SyscallReturn::SuccessU32(process.get_addresses().sram_start as u32),
+        2 => SyscallReturn::SuccessUsize(process.get_addresses().sram_start),
 
         // Op Type 3: Process memory end
-        3 => SyscallReturn::SuccessU32(process.get_addresses().sram_end as u32),
+        3 => SyscallReturn::SuccessUsize(process.get_addresses().sram_end),
 
         // Op Type 4: Process flash start
-        4 => SyscallReturn::SuccessU32(process.get_addresses().flash_start as u32),
+        4 => SyscallReturn::SuccessUsize(process.get_addresses().flash_start),
 
         // Op Type 5: Process flash end
-        5 => SyscallReturn::SuccessU32(process.get_addresses().flash_end as u32),
+        5 => SyscallReturn::SuccessUsize(process.get_addresses().flash_end),
 
         // Op Type 6: Grant region begin
-        6 => SyscallReturn::SuccessU32(process.get_addresses().sram_grant_start as u32),
+        6 => SyscallReturn::SuccessUsize(process.get_addresses().sram_grant_start),
 
         // Op Type 7: Number of defined writeable regions in the TBF header.
-        7 => SyscallReturn::SuccessU32(process.number_writeable_flash_regions() as u32),
+        7 => SyscallReturn::SuccessUsize(process.number_writeable_flash_regions()),
```

Comment:
I think this can't change. We can't have >2^32 regions work on one platform but not on another.

Reply:
I don't see why we couldn't, but u32 also seems large enough to me here.

Reply:
`usize` is currently the return type of `number_writeable_flash_regions`. Perhaps that should change if it needs to reflect a specific maximum number in the TBF header definition

Reply:
I would add there that the TBF format should not necessarily dictate the return type of a syscall (as other formats could feasibly exist), but instead the type should be chosen to best represent this thing. I think it is perfectly sensible, for instance, to promote a 32-bit field in the header to usize because it represents the size of some memory object and thus that is the right type.